lib/matplotlib/backends/qt_compat.py

```diff
@@ -216,3 +220,54 @@ def _setDevicePixelRatioF(obj, val):
     elif hasattr(obj, 'setDevicePixelRatio'):
         # Not available on Qt4 or some older Qt5.
         obj.setDevicePixelRatio(val)
+
+
+class _allow_interrupt:
+
+    def __init__(self, qApp):
+        self.interrupted_qobject = qApp
+        self.old_fd = None
+        self.old_sigint_handler = None
+        self.interrupt_caught = None
+
+        QAS = QtNetwork.QAbstractSocket
```

Comment:
Can / should we defer setting up the sockets until we know that we can safely register a signal handler?  In the case of julia binding into Python, the julia side handles all of the signals so maybe we should do nothing and leave this all to them?

Reply:
This comment was handled by protecting the call to the context manager and with the check at the top of the init.